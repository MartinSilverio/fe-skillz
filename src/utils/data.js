export const menuData = {
  menu: [
    {
      title: 'Lorem',
      id: 1,
      selected: false,
      children: [
        {
          title: 'Dolor',
          id: 11,
          selected: false,
          url: 'lorem/dolor',
        },
        {
          title: 'Ipsum',
          id: 12,
          selected: false,
          url: 'lorem/ipsum',
        },
        {
          title: 'Sit',
          id: 13,
          selected: false,
          url: 'lorem/sit',
        },
      ],
    },
    {
      title: 'Amet',
      id: 2,
      selected: false,
      url: 'amet',
    },
    {
      title: 'Lorem',
      id: 3,
      selected: false,
      children: [
        {
          title: 'Dolor',
          id: 31,
          selected: false,
          url: 'lorem/dolor',
        },
        {
          title: 'Ipsum',
          id: 32,
          selected: false,
          url: 'lorem/ipsum',
        },
        {
          title: 'Sit',
          id: 33,
          selected: false,
          url: 'lorem/sit',
        },
      ],
    },
  ],
};

// , consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris
