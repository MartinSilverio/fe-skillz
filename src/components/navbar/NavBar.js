import React from 'react';
import NavList from '../navlist/NavList';
import useWindowDimensions from '../../utils/hooks/useWidthDimensions';
import { MOBILE_BREAKPOINT } from '../../utils/constants';

import './NabBar.scss';
import logo from '../assets/we_are_wonderful.png';

function NavBar({ navItems }) {
  const { width } = useWindowDimensions();

  return (
    <nav id='navbar'>
      <img className='logo' alt='' src={logo}></img>
      {width > MOBILE_BREAKPOINT && (
        <NavList className='nav-list' navItems={navItems} />
      )}
    </nav>
  );
}

export default NavBar;
