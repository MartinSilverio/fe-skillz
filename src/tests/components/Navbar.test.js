import React from 'react';
import { shallow } from 'enzyme';
import NavBar from '../../components/navbar/NavBar';
import { menuData } from '../../utils/data';

describe('NavBar', () => {
  it('should display a navbar given data if window size is bigger than 800px', () => {
    expect.assertions(1);
    expect(shallow(<NavBar navItems={menuData.menu} />)).toMatchSnapshot();
  });
  it('should display header with button if window size is less than or equal to 800px', () => {
    expect.assertions(1);
    window.innerWidth = 800;
    expect(shallow(<NavBar navItems={menuData.menu} />)).toMatchSnapshot();
  });
});
