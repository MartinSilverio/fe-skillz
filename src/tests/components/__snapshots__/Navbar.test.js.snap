// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should display a navbar given data if window size is bigger than 800px 1`] = `
<Fragment>
  <nav
    aria-label="This Is Wonderful Navigation"
    id="navbar"
  >
    <img
      alt=""
      className="logo"
      src="we_are_wonderful.png"
    />
    <NavList
      aria-label="Top Level Navigation"
      className="nav-list"
      navItems={
        Array [
          Object {
            "children": Array [
              Object {
                "id": 11,
                "selected": false,
                "title": "Dolor",
                "url": "/lorem/dolor",
              },
              Object {
                "id": 12,
                "selected": false,
                "title": "Ipsum",
                "url": "/lorem/ipsum",
              },
              Object {
                "id": 13,
                "selected": false,
                "title": "Sit",
                "url": "/lorem/sit",
              },
            ],
            "id": 1,
            "selected": false,
            "title": "Lorem",
            "url": "/lorem",
          },
          Object {
            "id": 2,
            "selected": false,
            "title": "Amet",
            "url": "/amet",
          },
          Object {
            "children": Array [
              Object {
                "id": 31,
                "selected": false,
                "title": "Adipiscing",
                "url": "/consectetur/adipiscing",
              },
              Object {
                "id": 32,
                "selected": false,
                "title": "Eiusmod",
                "url": "/consectetur/eiusmod",
              },
              Object {
                "id": 33,
                "selected": false,
                "title": "Tempor",
                "url": "/consectetur/tempor",
              },
            ],
            "id": 3,
            "selected": false,
            "title": "Consectetur",
            "url": "/consectetur",
          },
        ]
      }
      role="menubar"
    />
  </nav>
</Fragment>
`;

exports[`NavBar should display header with button if window size is less than or equal to 800px 1`] = `
<Fragment>
  <nav
    aria-label="This Is Wonderful Navigation"
    id="navbar"
  >
    <img
      alt=""
      className="logo"
      src="we_are_wonderful.png"
    />
    <div
      className="nav-btn-container"
    >
      <button
        aria-label="open menu"
        className="nav-btn  false"
        onClick={[Function]}
      >
        MENU
      </button>
      <div
        className="triangle"
      />
    </div>
  </nav>
</Fragment>
`;
